integrations/influxdb_metrics_dashboard/lambda/upload_dashboard/main.go

```diff
@@ -0,0 +1,459 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/aws/aws-lambda-go/lambda"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/grafana"
+	grafanaTypes "github.com/aws/aws-sdk-go-v2/service/grafana/types"
+)
+
+func getWorkspaceByName(grafanaClient *grafana.Client, workspaceName string) (*grafanaTypes.WorkspaceSummary, error) {
+	const SleepDuration = 5
+	const MaxWaitIntervals = 200
+	waiterInterval := 0
+
+	resp, err := grafanaClient.ListWorkspaces(context.TODO(), &grafana.ListWorkspacesInput{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list workspaces: %v", err)
+	}
+
+	if resp == nil {
+		for waiterInterval < MaxWaitIntervals {
+			resp, err = grafanaClient.ListWorkspaces(context.TODO(), &grafana.ListWorkspacesInput{})
+			if resp != nil {
+				break
+			}
+			if err != nil {
+				return nil, fmt.Errorf("failed to list workspaces: %v", err)
+			}
+			waiterInterval++
+			time.Sleep(SleepDuration * time.Second)
+		}
+	}
+
+	if waiterInterval == MaxWaitIntervals {
+		return nil, fmt.Errorf("failed to find workspace %s in workspaces", workspaceName)
+	}
+
+	for _, workspace := range resp.Workspaces {
+		if *workspace.Name == workspaceName {
+			return &workspace, nil
+		}
+	}
+
+	return nil, fmt.Errorf("failed to find workspace %s in workspaces", workspaceName)
+}
+
+func getGrafanaHttpReq(requestType string, urlWithEndpoint string, payload io.Reader, serviceAccountTokenKey string) (*http.Request, error) {
+	req, err := http.NewRequest(requestType, "https://"+urlWithEndpoint, payload)
```

Comment:
The endpoint comes from the [WorkspaceSummary](https://github.com/aws/aws-sdk-go-v2/blob/main/service/grafana/types/types.go#L624) struct returned from `grafana.Client.ListWorkspaces`, so unless they change this logic, we can.

Reply:
I can't find any documentation or code notes that states it will or will not contain the scheme in the URL. Through my testing I can say that the URL does not contain the scheme when returned from `ListWorkspaces`.